src/xrdok.js

```diff
@@ -681,3 +681,130 @@ const XRiconchevron = 'xr-icon-chevron';
   });
 
 })();
+
+const XRplanegraph = 'xr-plane-graph';
+(function () {
+
+  AFRAME.registerComponent(XRplanegraph, {
+
+    schema: {
+      color: { type: 'color', default: '#CCF'},
+      colorMax: { type: 'color', default: '#FCC'},
+      colorMin: { type: 'color', default: '#CCF'},
+      descriptors: {type: 'array', default: [ '1960', '1970', '1980', '1990', '2000', '2010', '2020' ] },
+      primaryValues: { type: 'array', default: [ 6.8, 7, 7.5, 7, 8, 9, 9.5 ] },
+      primaryValuesBase: { type: 'number', default: 0},
+      secondaryValues: { type: 'array', default: [ 41, 40, 35, 36, 32, 28, 10 ] },
+      secondaryValuesBase: { type: 'number', default: 3.65 },
+      segmentLength: { type: 'number', default: 2 },
+      showAsLine: { type: 'boolean', default: false},
+      src: { type: 'string', default: './models/lufttemperatur_bbb_sortiert.csv'},
+    },
+
+    init: function() {
+      // data and element
+      var data = this.data;
+      var el = this.el;
+
+      let self = this;
+
+      // parse CSV, if available
+      Papa.parse(data.src, {  // eslint-disable-line no-undef
+        download: true,
+        complete: function(res) {
+          data.descriptors = [];
+          data.primaryValues = [];
+          data.secondaryValues = [];
+          for (var i = 0; i < res.data.length-1; i++) {
+            data.descriptors.push(res.data[i][0]);
+            data.primaryValues.push(parseFloat(res.data[i][1]/10));
+            data.secondaryValues.push(parseInt(res.data[i][2]));
+          }
+
+          self.geometry = new THREE.Geometry();
+          
+          if (data.showAsLine) {
+            self.createGeometryLine(el, self, data);
+          } else {
+            self.createGeometryPlane(el, self, data);
```

Comment:
this function is called on self, but also provides self as a parameter.

It would be better if you move this function out of this object and not call it on self. then you have to give it all dependencies as paramaters. like its already done here.